packages/react-dom-interactions/src/FloatingPortal.tsx

```diff
@@ -11,15 +11,19 @@ const DEFAULT_ID = 'floating-ui-root';
 export const FloatingPortal = ({
   children,
   id = DEFAULT_ID,
+  rootId = null,
 }: {
   children?: React.ReactNode;
   id?: string;
+  rootId?: string | null;
 }): React.ReactPortal | null => {
   const [mounted, setMounted] = useState(false);
   const portalRef = useRef<HTMLDivElement | null>(null);
 
   useLayoutEffect(() => {
-    const root = document.getElementById(id) as HTMLDivElement | null;
+    const rootNode = rootId ? document.getElementById(rootId) : null;
+    const root = (rootNode?.querySelector(`#${id}`) ??
+      document.getElementById(id)) as HTMLDivElement | null;
```

Comment:
I tested this code today, and it looks like it needs one small addition
```suggestion
    const root = (rootNode?.shadowRoot?.querySelector(`#${id}`) ??
      rootNode?.querySelector(`#${id}`) ??
      document.getElementById(id)) as HTMLDivElement | null;
```